confmap/expand.go

```diff
@@ -113,6 +114,9 @@ func (mr *Resolver) findAndExpandURI(ctx context.Context, input string) (any, bo
 	if err != nil {
 		return input, false, err
 	}
+	if !changed {
+		return input, false, nil
+	}
 	repl, err := toString(uri, expanded)
```

Comment:
Got confused why `uri` is needed, so created a small cleanup [PR](https://github.com/open-telemetry/opentelemetry-collector/pull/10244)